### PR TITLE
refactoring

### PR DESCRIPTION
**Pull request by arszp10**

### CODE REVIEW

**Comment by kuzkok** [comment]
Hi, thanks for PR. 
Before merge some work should be done:
- Properties functions should be written in one place. Create mixin as described in [**includes** help sections for Class](http://leafletjs.com/reference.html#class), and include it to Marker and Path classes.
- getPropertiesId function is unneccesary
- getProperties should be renamed to getProperty
- There are spec's must be written for new functions

File: src/Properties.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var PropertiesMixin = {
+  setProperties: function(obj) {
+    for (var i in obj) {
```
Comment (kuzkok) [inline src/Properties.js:3]: need check with hasOwnProperty

File: spec/WFST.js [hunk 2]
```diff
@@ -1,14 +1,14 @@
 /**
  * Created by PRadostev on 18.02.2015.
  */
-describe('WFST', function () {
```
Comment (kuzkok) [inline spec/WFST.js:4]: please restore this spaces between `function` and `(`. They presented in all project.

File: src/Properties.js [hunk 3]
```diff
@@ -1,11 +1,27 @@
 var PropertiesMixin = {
   setProperties: function(obj) {
     for (var i in obj) {
-      this.feature.properties[i] = obj[i];
+      if (this.feature.properties.hasOwnProperty(i)) {
```
Comment (kuzkok) [inline src/Properties.js]: check should be for obj.hasOwnProperty

File: src/Properties.js [hunk 4]
```diff
@@ -1,11 +1,27 @@
...
+        this.feature.properties[i] = obj[i];
+      }
     }
   },
   getProperty: function(field) {
     return this.feature.properties[field];
+  },
+  addProperties: function(obj){
```
Comment (kuzkok) [inline src/Properties.js]: unnecessary, dublicate code from  setProperties

File: src/Properties.js [hunk 5]
```diff
@@ -1,11 +1,27 @@
...
+    for (var i in obj) {
+      if (!this.feature.properties.hasOwnProperty(i)) {
+        this.feature.properties[i] = obj[i];
+      }
+    }
+  },
+  deleteProperty: function(arr){
+    for (var i = 0; i < arr.length; i++) {
+      if (this.feature.properties.hasOwnProperty(arr[i])) {
+        delete this.feature.properties[arr[i]];
```
Comment (kuzkok) [inline src/Properties.js]: may be ```this.feature.properties[arr[i]]=null``` will be better solution

File: src/Properties.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
+var PropertiesMixin = {
+  setProperties: function (obj) {
+    for (var i in obj) {
+      if (this.feature.properties.hasOwnProperty(i)) {
```
Comment (kuzkok) [inline src/Properties.js]: check should be for obj.hasOwnProperty

File: src/Properties.js [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        this.feature.properties[i] = obj[i];
+      }
+    }
+  },
+  getProperty: function (field) {
+    return this.feature.properties[field];
+  },
+  addProperties: function (obj) {
```
Comment (kuzkok) [inline src/Properties.js]: unnecessary function, all this work can be done with setProperties, must be removed.

File: src/Properties.js [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    for (var i in obj) {
+      if (!this.feature.properties.hasOwnProperty(i)) {
+        this.feature.properties[i] = obj[i];
+      }
+    }
+  },
+  deleteProperty: function (arr) {
+    for (var i = 0; i < arr.length; i++) {
+      if (this.feature.properties.hasOwnProperty(arr[i])) {
+        delete this.feature.properties[arr[i]];
```
Comment (kuzkok) [inline src/Properties.js:15]: `this.feature.properties[arr[i]]=null` will be better solution, need rename function to clearProperties()

File: src/Properties.js (hunk 6 above)
Comment (arszp10) [reply src/Properties.js]: sorry
i'm ashamed for this error

File: src/Properties.js (hunk 8 above)
Comment (arszp10) [reply src/Properties.js:15]: > this.feature.properties[arr[i]]=null will be better solution

why?

Comment (kuzkok) [reply src/Properties.js:15]: Particular example, when you work with WFST, is a set property to null. 
But with setProperties this function also unnecessary, all this this work can be done with

``` javascript
setProperties({a: null});
```

I was wrong, sometimes programmer should can remove properties from wfs:Transaction request, and deleteProperties can help.
Please restore function to you first version, but it should be renamed to deleteProperties.

Comment (arszp10) [reply src/Properties.js:15]: do you want following code:

``` javascript
deleteProperties: function (arr) {
    for (var i = 0; i < arr.length; i++) {
      if (this.feature.properties.hasOwnProperty(arr[i])) {
        delete this.feature.properties[arr[i]];
      }
    }
  }
```

?

Comment (kuzkok) [reply src/Properties.js:15]: Yes. And clearProperties() function should be removed.

**Pull request merged**